lgc/include/lgc/state/PalMetadata.h

```diff
@@ -147,17 +147,38 @@ class PalMetadata {
   // Erase the color export info
   void eraseColorExportInfo();
 
-  // Finalize PAL metadata for pipeline.
-  // TODO Shader compilation: The idea is that this will be called at the end of a pipeline compilation, or in
-  // an ELF link, but not at the end of a shader/half-pipeline compile.
-  void finalizePipeline();
+  // Finalize PAL metadata for pipeline, part-pipeline or shader compilation.
+  void finalizePipeline(bool isWholePipeline);
 
   // Updates the PS register information that depends on the exports.
   void updateSpiShaderColFormat(llvm::ArrayRef<ColorExportInfo> exps, bool hasDepthExpFmtZero, bool killEnabled);
 
   // Sets the finalized 128-bit cache hash.  The version identifies the version of LLPC used to generate the hash.
   void setFinalized128BitCacheHash(const lgc::Hash128 &finalizedCacheHash, const llvm::VersionTuple &version);
 
+  // Store the fragment shader input mapping information for a fragment shader being compiled by itself (partial
+  // pipeline compilation).
+  void addFragmentInputInfo(llvm::ArrayRef<unsigned> data1, llvm::ArrayRef<unsigned> data2);
```

Comment:
I would not know how to call this function. How are `data1` and `data2` related? Is this a map decomposed into two arrays (keys and values)? Do these arrays have to be of the same lengths?

Would it be possible to add an example in the comment?

Reply:
See Steven's comment. I'll mark this one resolved this one so discussion can happen over there.